README.md

```diff
@@ -3,14 +3,138 @@
+
+In its operation, the action will manage a tag (named `discourse-gatekeeper/content`), 
+that represents the last synced content with Discourse. Please do not remove this tag to ensure 
+correct execution of the action. 
+
 ## Getting Started
 
+The action can be included in your CI/CD pipeline by adding the following step
+
+```yaml
+  - name: Publish documentation
+    uses: canonical/discourse-gatekeeper@stable
+    id: publishDocumentation
+    with:
+      discourse_host: discourse.charmhub.io
+      discourse_api_username: ${{ secrets.DISCOURSE_API_USERNAME }}
+      discourse_api_key: ${{ secrets.DISCOURSE_API_KEY }}
+      github_token: ${{ secrets.GITHUB_TOKEN }}
+      dry_run: "true" // "false" 
+```
+
+This action requires an API username and key to discourse. For Canonical
+staff, please file a ticket with IS to request one. Note that there is a
+rate limit on the number of topics that can be created by a user per day on
+discourse. 
+
+There is a parameter, `dry_run`, which will do everything except
+make changes on discourse and log what would have happened. This is the recommended
+use to enable one-way sync where the content in the repository is kept in sync with 
+Discourse but where no upload is allowed.
+
+> :warning: Two-way sync between Discourse and Github is currently not allowed and 
+> can only be used with leadership approval.
+
+### Permissions
+
+Make sure that the action runs in workflow with correct permission settings. In 
+particular, the action should be allowed to:
+1. Edit content and push commits/branches/tags in order to commit community contributions 
+   to dedicated branches, e.g. `discourse-gatekeeper/migrate` , as well update the position 
+   of the `discourse-gatekeeper/content` tag. 
+2. Open/amend/close pull-requests in order to raise PR with community contributions
+
+Thus, when the action is embedded in an external workflow, make sure that you pass 
+credentials and permissions accordingly, e.g. 
+
+```shell
+  sync-docs:
+    uses: ./.github/workflows/sync_docs.yaml
+    secrets: inherit
+    permissions:
+      contents: write
+      pull-requests: write
+```
+
+### Recommended Setup
+
+Although the action is designed to always perform **Raise, Check and Upload** functions,
+for best user-experience, we suggest the action to be included in:
+
+1. Scheduled workflows with dry-run enabled: to make sure PR with community contributions
+   are raised regularly
+2. CI checks with dry-run enabled: to make sure amends to documentations are not conflicting with community 
+   contributions, as well as increases the rate at which PR with community contributions are raised
+3. Release pipelines: to upload *(experimental)* documentations edits to Discourse and make sure 
+   that the `discourse-gatekeeper/content` tag is updated regularly 
+
+### Disclaimers
+ 
+1. The action is currently in alpha state. If you encounter this issue, the action will fail and report
```

Comment:
Is  `If you encounter this issue` referring to anything in particular? It doesn't seem to be clear from the context